### ISSUE TITLE
AWS CLI v2 aws glacier get-job-output fails if job ID starts with hyphen

### ISSUE THREAD

**Issue by phistrom** [aws/aws-cli#5985]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [X] I've gone though the [User Guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html) and the [API reference](https://docs.aws.amazon.com/cli/latest/reference/)
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-cli/issues) and didn't find any solution
  
**Describe the bug**
The following command gives the error `aws.exe: error: argument --job-id: expected one argument`
It seems to be related to the fact that the job ID starts with a hyphen.
```
aws glacier get-job-output --vault-name Some-Vault --account-id - --job-id "-0Bl0xNXREb7Y-xJAdqPitPZ28ehLPom-Gd_8m5aTyoujAtu-MlKuaBqZQh1oAzGj7GIcXNMUjfssKg0vSA-CLwr1X8B" output.json
```

**SDK version number**
`aws-cli/2.1.27 Python/3.7.9 Windows/10 exe/AMD64 prompt/off`

**Platform/OS/Hardware/Device**
Windows 10 Pro 20H2 x64

**To Reproduce (observed behavior)**
See above command

**Expected behavior**
For the CLI to not complain about a missing argument

**Logs/output**
Get full traceback and error logs by adding `--debug` to the command.
```
2021-03-02 21:40:25,801 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/2.1.27 Python/3.7.9 Windows/10 exe/AMD64
2021-03-02 21:40:25,801 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['glacier', 'get-job-output', '--debug', '--vault-name', 'Some-Vault', '--account-id', '-', '--job-id', '-0Bl0xNXREb7Y-xJAdqPitPZ28ehLPom-Gd_8m5aTyoujAtu-MlKuaBqZQh1oAzGj7GIcXNMUjfssKg0vSA-CLwr1X8B']
2021-03-02 21:40:25,847 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function add_s3 at 0x0000014EABBFB3A8>
2021-03-02 21:40:25,847 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function add_ddb at 0x0000014EAB266F78>
2021-03-02 21:40:25,847 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <bound method BasicCommand.add_command of <class 'awscli.customizations.configure.configure.ConfigureCommand'>>
2021-03-02 21:40:25,847 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function change_name at 0x0000014EAB205EE8>
2021-03-02 21:40:25,847 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function change_name at 0x0000014EAB20F678>
2021-03-02 21:40:25,847 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function alias_opsworks_cm at 0x0000014EABC0C3A8>
2021-03-02 21:40:25,848 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function add_history_commands at 0x0000014EAB2B14C8>
2021-03-02 21:40:25,848 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <bound method BasicCommand.add_command of <class 'awscli.customizations.devcommands.CLIDevCommand'>>
2021-03-02 21:40:25,848 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function add_waiters at 0x0000014EABC0CA68>
2021-03-02 21:40:25,848 - MainThread - botocore.loaders - DEBUG - Loading JSON file: C:\Program Files\Amazon\AWSCLIV2\awscli\data\cli.json
2021-03-02 21:40:25,850 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <function resolve_types at 0x0000014EAB369708>
2021-03-02 21:40:25,850 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <function no_sign_request at 0x0000014EAB3699D8>
2021-03-02 21:40:25,850 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <function resolve_verify_ssl at 0x0000014EAB369948>
2021-03-02 21:40:25,850 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <function resolve_cli_read_timeout at 0x0000014EAB369AF8>
2021-03-02 21:40:25,850 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <function resolve_cli_connect_timeout at 0x0000014EAB369A68>
2021-03-02 21:40:25,850 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <built-in method update of dict object at 0x0000014EABCB6E58>
2021-03-02 21:40:25,851 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/2.1.27 Python/3.7.9 Windows/10 exe/AMD64 prompt/off
2021-03-02 21:40:25,851 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['glacier', 'get-job-output', '--debug', '--vault-name', 'Some-Vault', '--account-id', '-', '--job-id', '-0Bl0xNXREb7Y-xJAdqPitPZ28ehLPom-Gd_8m5aTyoujAtu-MlKuaBqZQh1oAzGj7GIcXNMUjfssKg0vSA-CLwr1X8B']
2021-03-02 21:40:25,852 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_timestamp_parser at 0x0000014EABBFBF78>
2021-03-02 21:40:25,852 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function register_uri_param_handler at 0x0000014EAB008048>
2021-03-02 21:40:25,852 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_binary_formatter at 0x0000014EABC6D828>
2021-03-02 21:40:25,852 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function no_pager_handler at 0x0000014EAAFAADC8>
2021-03-02 21:40:25,853 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_assume_role_provider_cache at 0x0000014EAB02DB88>
2021-03-02 21:40:25,854 - MainThread - botocore.utils - DEBUG - IMDS ENDPOINT: http://169.254.169.254/
2021-03-02 21:40:25,855 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function attach_history_handler at 0x0000014EAB2B1558>
2021-03-02 21:40:25,855 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_json_file_cache at 0x0000014EAB257EE8>
2021-03-02 21:40:25,900 - MainThread - botocore.loaders - DEBUG - Loading JSON file: C:\Program Files\Amazon\AWSCLIV2\botocore\data\glacier\2012-06-01\service-2.json
2021-03-02 21:40:25,903 - MainThread - botocore.hooks - DEBUG - Event building-command-table.glacier: calling handler <function add_waiters at 0x0000014EABC0CA68>
2021-03-02 21:40:25,940 - MainThread - botocore.loaders - DEBUG - Loading JSON file: C:\Program Files\Amazon\AWSCLIV2\botocore\data\glacier\2012-06-01\waiters-2.json
2021-03-02 21:40:25,940 - MainThread - awscli.clidriver - DEBUG - OrderedDict([('account-id', <awscli.arguments.CLIArgument object at 0x0000014EABDA7A48>), ('vault-name', <awscli.arguments.CLIArgument object at 0x0000014EABDA7B08>), ('job-id', <awscli.arguments.CLIArgument object at 0x0000014EABDA7B88>), ('range', <awscli.arguments.CLIArgument object at 0x0000014EABDA7B48>)])
2021-03-02 21:40:25,941 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.glacier.get-job-output: calling handler <function add_streaming_output_arg at 0x0000014EABC04168>
2021-03-02 21:40:25,943 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.glacier.get-job-output: calling handler <function add_cli_input_json at 0x0000014EAB034DC8>
2021-03-02 21:40:25,943 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.glacier.get-job-output: calling handler <function add_cli_input_yaml at 0x0000014EAB0345E8>
2021-03-02 21:40:25,944 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.glacier.get-job-output: calling handler <function unify_paging_params at 0x0000014EAB26E678>
2021-03-02 21:40:25,988 - MainThread - botocore.loaders - DEBUG - Loading JSON file: C:\Program Files\Amazon\AWSCLIV2\botocore\data\glacier\2012-06-01\paginators-1.json
2021-03-02 21:40:25,988 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.glacier.get-job-output: calling handler <function add_generate_skeleton at 0x0000014EAB361708>
2021-03-02 21:40:25,989 - MainThread - awscli.clidriver - DEBUG - Exception caught in main()
Traceback (most recent call last):
  File "argparse.py", line 1787, in parse_known_args
  File "argparse.py", line 1993, in _parse_known_args
  File "argparse.py", line 1923, in consume_optional
  File "argparse.py", line 2088, in _match_argument
argparse.ArgumentError: argument --job-id: expected one argument

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "lib\site-packages\awscli\clidriver.py", line 457, in main
  File "lib\site-packages\awscli\clidriver.py", line 586, in __call__
  File "lib\site-packages\awscli\clidriver.py", line 721, in __call__
  File "lib\site-packages\awscli\argparser.py", line 209, in parse_known_args
  File "lib\site-packages\awscli\argparser.py", line 95, in parse_known_args
  File "argparse.py", line 1794, in parse_known_args
  File "lib\site-packages\awscli\argparser.py", line 128, in error
awscli.argparser.ArgParseException: usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help

aws: error: argument --job-id: expected one argument


usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help

aws: error: argument --job-id: expected one argument
```
**Additional context**
Now that I see it with the `--debug` flag, it's using argparse under the hood. [Looks like it will work with an = sign.](https://stackoverflow.com/a/52636344/489667)
`aws glacier get-job-output --vault-name Some-Vault --account-id - --job-id=-0Bl0xNXREb7Y-xJAdqPitPZ28ehLPom-Gd_8m5aTyoujAtu-MlKuaBqZQh1oAzGj7GIcXNMUjfssKg0vSA-CLwr1X8B output.json`

**Comment by stobrien89** [comment]
Hi @phistrom, 

Thanks for writing in!  Unfortunately, this is an [issue](https://bugs.python.org/issue9334) with the stdlib argument parsing library we use— we've been waiting on a fix for quite some time.  Closing for now, as this is a duplicate of [#1135](https://github.com/aws/aws-cli/issues/1135).
